src/prefect/utilities/edges.py

```diff
@@ -0,0 +1,118 @@
+import prefect
+from typing import Dict, Any
+
+
+class EdgeAnnotation:
+    """
+    A parent class for containers like `unmapped` and `flatten` that indicate
+    an annotation for an edge
+
+    EdgeAnnotations store a set of annotations, allowing multiple annotations
+    to be applied.
+    """
+
+    def __init__(self, task: "prefect.Task", annotations: Dict[str, Any]):
+        self.annotations = dict()  # type: Dict[str, Any]
+        if isinstance(task, EdgeAnnotation):
+            self.annotations.update(task.annotations)
+        self.annotations.update(annotations)
+        self.task = prefect.utilities.tasks.as_task(task)
+
+
+class mapped(EdgeAnnotation):
+    """
+    A container for specifying that a task should be mapped over when supplied as
+    the input to another task.
+
+    Args:
+        - task (Task): the task to mark as "unmapped"; if not a Task subclass,
+            Prefect will attempt to convert it to one.
+
+    Example:
+        ```python
+        from prefect import Flow, Task, mapped
+
+        class AddTask(Task):
+            def run(self, x, y):
+                return x + y
+
+        class ListTask(Task):
+            def run(self):
+                return [1, 2, 3]
+
+        with Flow("My Flow"):
+            add = AddTask()
+            ll = ListTask()
+            result = add(x=mapped(ll), y=5)
+        ```
+    """
+
+    def __init__(self, task: "prefect.core.task.Task"):
```

Comment:
This type annotation isn't really correct, since `task` could also be an `EdgeAnnotation` object, which isn't a `Task`.